### DISCUSSION TITLE
XCharts-3.8.0 - New Input System & Text Mesh Pro Library Rebuilt issue

### DISCUSSION THREAD

**Discussion by Aunkon** [XCharts-Team/XCharts#270] [Q&A]
This problem is occurring while our project newly cloning or deleted the Library Folder.
Can anyone help with this issue?

![image](https://github.com/XCharts-Team/XCharts/assets/19836171/d89f1c11-45e6-418a-8b30-c53af4b672d0)

This issue is solved by removing these two define symbols. And Enabling the Input System and Text Mesh Pro again.

![Screenshot 2023-08-03 120742](https://github.com/XCharts-Team/XCharts/assets/19836171/258b3047-4322-414f-90aa-2691072add37)
![image](https://github.com/XCharts-Team/XCharts/assets/19836171/a0f02445-41c2-4f4a-bd66-77e1b27f417c)
![Screenshot 2023-08-03 121118](https://github.com/XCharts-Team/XCharts/assets/19836171/94df7ebf-a1c6-406e-9f28-90918a15fd02)
![image](https://github.com/XCharts-Team/XCharts/assets/19836171/ebced9c3-4f82-470a-b390-f041a58755c0)

Now the Final Problem we are fetching is.
### **The reason we can't fix our CI-CD cause is Build the Library Cache while CI Building.**
Can you please solve this issue ASAP on the next version or provide any other Solution to fix our Project cloning?
### **Note: We must have to use Unity New Input System and Text Mesh Pro Because we have thousands of dependencies in our projects.**

**Comment by monitor1394** [comment]
It is true that this problem has always existed, and no suitable solution has been found yet.

Because the package on the repository does not have a reference to TMP and NewInputSystem by default.

When you enable TMP and NewInputSystem locally, the package file is overwritten, but the script symbol in PlayerSetting is still there, so an error is reported.

Maintaining more than one branch can solve this problem, but it feels a bit cumbersome.

Do you have any suggestions?

**Reply by Aunkon** [reply]
Your concern about my problem is really very appreciable. I am trying to find the Solution. But can't found till now.
Have an Idea, though I am not sure if it should work or not. I am also so sorry that I am limited to thinking that I still can't explore the Full XCharts Packages till now. Therefore the Suggestion of mine's are - 
- Could it be solved by Shifting the TMP and NewInputSystem Configuration File inside the Project's Assets/XCharts/Resources Folder as XCharts Configuration File as Like Import XCharts Essential Files? And Keeps the XCharts Configuration inside the whole project's Source codes inside version controlling?
- Or Can it be a separate package to work the XCharts with TMP and NewInputSystem? Though Unity by Default using the TMP now. I know this is a very dumb idea.

**Reply by Aunkon** [reply]
- Or We can check whether the project Initialization from a fresh opening or the XCharts Package importing Check whether Unity Project's Configuration is the TMP and New InputSystem Activate or not inside Project Settings by **Scripting Define Symbols** or the **Active Input Handling***, Input System Package (New) is active or not, The XCharts Configuration will be set to TMP Activate and New InputSystem Active, by default?

**Comment by monitor1394** [comment]
I added a [XCharts-Daemon](https://github.com/XCharts-Team/XCharts-Daemon) package to solve this problem. If it is convenient for you, you can test whether it works properly in the CI-CD environment.

**Reply by Aunkon** [reply]
Sorry for the late reply, I am grateful for your concern. Most of the issues are solved after adding [XCharts-Daemon](https://github.com/XCharts-Team/XCharts-Daemon),
But still, these issues remained while Initializing the Project from Git again.

![Screenshot 2023-08-24 153511](https://github.com/XCharts-Team/XCharts/assets/19836171/b00f23d4-059d-4a7a-9204-434706018b73)

**Reply by monitor1394** [reply]
I tested the following situations and they all worked:

1. Delete the Library directory
2. Remove XCharts from the PackageManager and add them again
3. Update XCharts when it is placed in the Assets directory

You need to make sure that there are no compilation errors other than those associated with XCharts.

The XChartsDaemon changes asmdef and recompiles XCharts when it is re-imported.

**Reply by monitor1394** [reply]
![image](https://github.com/XCharts-Team/XCharts/assets/1450329/ebd8b6c0-c7e7-41a8-8a45-42554d04c506)

**Reply by Aunkon** [reply]
https://youtu.be/JJV6k0MnDZE
Maybe can't properly try, but I found the issues remain the same as in this video.
Also, the Step 2 & 3 will be difficult to manage in CI-CD.
You may have known that We can only perform [Unity Command Line](https://docs.unity3d.com/2023.1/Documentation/Manual/EditorCommandLineArguments.html) arguments in CI-CD.

**Reply by monitor1394** [reply]
Thank you for your detailed feedback on the video.
I will re-test it according to the steps in the video.

**Comment by monitor1394** [comment]
Try to update XCharts-Daemon to version 1.0.1, it works well in my tests.

**Reply by Aunkon** [reply]
I am sorry, But the issue still remain in my project.
Note:- My Unity Version is 2023.1.0f1
![Screenshot 2023-08-26 052939](https://github.com/XCharts-Team/XCharts/assets/19836171/f4780427-0eb2-4e77-a25d-9d68ed93ef51)
